src/lib/nats/shutdown.ts

```diff
@@ -0,0 +1,10 @@
+import { drain } from "./nats";
+import { NatsConnection } from "nats";
+
+export default async function shutdown(
+    exitCode: number,
+    natsConnection: NatsConnection
+) : Promise<void> {
+    await drain(natsConnection);
+    process.exit(exitCode);
```

Comment:
imo process management should not happen in library code

Reply:
Do you think that I should move it alongside the init function ?

Reply:
let's discuss how we want to handle this across microservices and plan for refactor